apptest.go

```diff
@@ -177,6 +180,52 @@ func (a *AppSetup) InstallApps(ctx context.Context, apps []App) error {
 	return nil
 }
 
+func (a *AppSetup) UpgradeApp(ctx context.Context, current, desired App) error {
+	var err error
+
+	err = a.createAppCatalogs(ctx, []App{current, desired})
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	// if current version have no specific version, use the latest instead.
+	if current.Version == "" && current.SHA == "" {
+		catalogURL, err := getCatalogURL(current)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		version, err := appcatalog.GetLatestVersion(ctx, catalogURL, current.Name, "")
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		current.Version = version
+	}
+
+	err = a.createApps(ctx, []App{current})
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = a.waitForDeployedApps(ctx, []App{current})
```

Comment:
Why do we use waitForDeployedApps here? Can we use waitForDeployedApp instead?

Reply:
Wow, I didn't know that it is there! Thanks for pointing that out.